attribute-service-impl/src/main/java/org/hypertrace/core/attribute/service/AttributeServiceImpl.java

```diff
@@ -310,26 +315,49 @@ public void findAll(Empty request, StreamObserver<AttributeMetadata> responseObs
     }
   }
 
+  @Override
+  public void getAttributes(
+      GetAttributesRequest request, StreamObserver<GetAttributesResponse> responseObserver) {
+    String tenantId = RequestContext.CURRENT.get().getTenantId().orElse(null);
```

Comment:
Maybe this is never going to happen but no nil checks needed here? Also, this looks a bit opposite to [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter)

Reply:
No nil checks needed, this is a method invoked by GRPC and the args can never be null. As for the tenant ID, I tend to agree, but it's debatable and this is one of several methods in the service so I valued consistency more.

My preferred style is to just pull the request context and pass it on to the downstream code. I'd guess the counterarguments that would be raised here would look like:
- fail fast if the request is missing a tenant id. IMO this is meaningful - we don't want to do a bunch of work that doesn't require the tenant id before realizing we can't complete (assuming we don't have transactionality, which would make this even more important)
- simplify the downstream code so it doesn't need its own checks everywhere (this is kind of the whole point though, so this argument I don't buy)